### PR TITLE
[validation] Aggregate loss by valid token count

### PR DESCRIPTION
**Pull request by taking-lying-flat**
## Summary

- accumulate raw validation loss sums and global valid-token counts across batches
- normalize once after the validation loop, after reducing the total loss across the loss mesh

## Root cause

Validation normalized each batch loss by that batch's valid-token count and then averaged those normalized losses equally across validation steps. This computes a mean of batch means rather than the intended global token mean, so batches with fewer valid tokens receive disproportionate weight.

For batches with `(loss_sum, valid_tokens)` values `(2, 1)` and `(90, 9)`, the previous result was `(2 / 1 + 90 / 9) / 2 = 6.0`. The correct token-weighted result is `(2 + 90) / (1 + 9) = 9.2`.

## Impact

Validation cross-entropy now reflects the average over all non-ignored tokens. This is especially important for SFT datasets where prompt tokens use `IGNORE_INDEX`, response lengths vary, or the last batch is incomplete.

### CODE REVIEW

File: torchtitan/components/validate.py [hunk 1]
```diff
@@ -327,18 +330,19 @@ def validate(
                     predictions = model_parts[0](inputs, **extra_kwargs)
                     loss_sum, _ = self.loss_fn(predictions, labels)
 
-            accumulated_losses.append(loss_sum.detach() / global_valid_tokens)
+            accumulated_loss_sums.append(loss_sum.detach())
```
Comment (tianyu-l) [inline torchtitan/components/validate.py]: could you follow https://github.com/pytorch/torchtitan/pull/4146 and do this inplace?

**Review by tianyu-l — COMMENTED** [review]
thanks for catching, one comment

File: torchtitan/components/validate.py (hunk 1 above)
Comment (taking-lying-flat) [reply torchtitan/components/validate.py]: lgtm, thanks

**Review by tianyu-l — APPROVED** [review]
